src/Telemetry/AzureDevOpsTimeline/AzureDevOpsTimeline.cs

```diff
@@ -382,6 +414,104 @@ private static string GetBucket(AugmentedTimelineIssue augIssue)
             return await azureServer.ListBuilds(project, cancellationToken, minDateTime, limit);
         }
 
+        private async Task<BuildPath> GetLongestBuildPath(MaestroBuild build, List<string> visitedRepositories, Build[] builds, AzureDevOpsTimelineOptions options)
```

Comment:
>GetLongestBuildPath [](start = 38, length = 19)

This is a recursive function where each recursion had a Kusto query and SEVERAL SQL queries. I think this needs to be redesigned to avoid so much dipping into the data sources.  We should only have to make 1 SQL query and 1 Kusto query to get all the data when we need it.